src/lib.rs

```diff
@@ -45,7 +46,10 @@ pub trait Body {
     type Error;
 
     /// Attempt to pull out the next data buffer of this stream.
-    fn poll_data(&mut self) -> Poll<Option<Self::Data>, Self::Error>;
+    fn poll_data(
+        self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+    ) -> Poll<Result<Option<Self::Data>, Self::Error>>;
```

Comment:
`Stream` puts the `Result` inside the `Option`, and there's a `Try` for `Poll<Option<Result>>`, so we should probably keep the same order.

Reply:
Ah yes, I missed that :+1:

Reply:
Yeah....totally agree, I kinda just bruteforced myself through this. Will update it.